pkg/webhook/workspace/mutating_cfg.go

```diff
@@ -29,72 +29,111 @@ func buildMutateWebhookCfg() *v1beta1.MutatingWebhookConfiguration {
+					Resources:   []string{"workspaces"},
 				},
-				Rules: []v1beta1.RuleWithOperations{
-					{
-						Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
-						Rule: v1beta1.Rule{
-							APIGroups:   []string{"workspace.che.eclipse.org"},
-							APIVersions: []string{"v1alpha1"},
-							Resources:   []string{"workspaces"},
-						},
-					},
+			},
+		},
+	}
+
+	workspaceObjMutateWebhook := v1beta1.MutatingWebhook{
+		Name:          "mutate-ws-resources.che-workspace-controller.svc",
+		FailurePolicy: &mutateWebhookFailurePolicy,
+		ClientConfig: v1beta1.WebhookClientConfig{
+			Service: &v1beta1.ServiceReference{
+				Name:      "workspace-controller",
+				Namespace: "che-workspace-controller",
+				Path:      &mutateWebhookPath,
+			},
+			CABundle: server.CABundle,
+		},
+		ObjectSelector: &metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				{
+					Key:      config.WorkspaceIDLabel,
+					Operator: labelExistsOp,
 				},
 			},
+		},
+		MatchPolicy: &equivalentMatchPolicy,
+		Rules: []v1beta1.RuleWithOperations{
 			{
-				Name:          "mutate-ws-resources.che-workspace-controller.svc",
-				FailurePolicy: &mutateWebhookFailurePolicy,
-				ClientConfig: v1beta1.WebhookClientConfig{
-					Service: &v1beta1.ServiceReference{
-						Name:      "workspace-controller",
-						Namespace: "che-workspace-controller",
-						Path:      &mutateWebhookPath,
-					},
-					CABundle: server.CABundle,
+				Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
+				Rule: v1beta1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"pods"},
 				},
-				ObjectSelector: &metav1.LabelSelector{
-					MatchExpressions: []metav1.LabelSelectorRequirement{
-						{
-							Key:      config.WorkspaceIDLabel,
-							Operator: labelExistsOp,
-						},
-					},
+			},
+			{
+				Operations: []v1beta1.OperationType{v1beta1.Update},
+				Rule: v1beta1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"services"},
 				},
-				MatchPolicy: &equivalentMatchPolicy,
-				Rules: []v1beta1.RuleWithOperations{
-					{
-						Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
-						Rule: v1beta1.Rule{
-							APIGroups:   []string{""},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"pods"},
-						},
-					},
-					{
-						Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
-						Rule: v1beta1.Rule{
-							APIGroups:   []string{"apps"},
-							APIVersions: []string{"v1"},
-							Resources:   []string{"deployments"},
-						},
-					},
+			},
+			{
+				Operations: []v1beta1.OperationType{v1beta1.Create, v1beta1.Update},
+				Rule: v1beta1.Rule{
+					APIGroups:   []string{"apps"},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"deployments"},
 				},
 			},
+			{
+				Operations: []v1beta1.OperationType{v1beta1.Update},
```

Comment:
Do you think it makes sense to deny creating other workspace-related objects for everyone except creator and controller?
For sure except pods - since they are managed by replicasSets...

Reply:
I think it makes sense; added in https://github.com/che-incubator/che-workspace-operator/pull/89/commits/aa9415d6364d8abb21cfe92dd252b5f351eb1b03